mpqp/core/instruction/gates/native_gates.py

```diff
@@ -598,6 +608,51 @@ def __repr__(self) -> str:
         2
     )
 
+    def to_matrix(self, nb_qubits: int = 0) -> npt.NDArray[np.complex64]:
+        """Constructs the matrix representation of a SWAP gate for two qubits in
+        a multi-qubit system.
+
+        Args:
+            nb_qubits: The total number of qubits in the system. If not
```

Comment:
in the circuit ?

Reply:
done

Reply:
It's not necessarily the full size of the circuit, maybe we could rephrase this as 
```
Desired size of the gate (in term of qubits)
```

Reply:
i put:  nb_qubits: The total number for qubits gate representation.